src/main/java/backend/Logic.java

```diff
@@ -85,22 +85,26 @@ public void refresh() {
     }
 
     public CompletableFuture<Boolean> openPrimaryRepository(String repoId) {
-        return openRepository(repoId, true);
+        assert Utility.isWellFormedRepoId(repoId);
+        prefs.setLastViewedRepository(repoId);
+        if ((isAlreadyOpen(repoId) && repoId.equals(getDefaultRepo())) || models.isRepositoryPending(repoId)) {
```

Comment:
if i understand this correctly, the extra check here is the main change in this pr. so that we don't do anything if the primary repo remains the same right? my question is will there ever be a case where this method is called but the primary repo isn't changed?